cpp/src/arrow/util/compression_zlib.cc

```diff
@@ -392,40 +395,48 @@ class GZipCodec : public Codec {
       return 0;
     }
 
-    // Reset the stream for this block
-    if (inflateReset(&stream_) != Z_OK) {
-      return ZlibErrorPrefix("zlib inflateReset failed: ", stream_.msg);
-    }
+    // inflate() will not automatically decode concatenated gzip members, keep calling
```

Comment:
is the comment right here? Seems this means continue `inflateResult` and run one or multiple `inflate`?

Reply:
yes this means to continue calling inflate and concatenating inflate results until reading(inflatting) all input data
I put it on the while condition to epxplain the reason behind it
Do you think it should be on the inflate call? or somwhere else?

Reply:
Hmmm yeah I think it should be on the inflate call, because here it's ambigious...

Reply:
I think I will do:
`//loop until all input bytes are readen
while(read_input_bytes < input_length)
...
// inflate 1 gzip member each loop iteration
inflate(...)
`

what do you think?

Reply:
Maybe comment the `inflate` and `inflateInit` is ok?

Reply:
I didn't comment the inflateInit

Reply:
```
loop {
  inflateReset
  loop {
    inflate
  }
}
```

Seems we'd better separate comment `inflateReset` and `inflate` around there code? And the `inflate` logic is unchanged here( it might call `inflate` multiple times before ). So I didn't know why the comment is here...

My wording is wrong here, it's `inflateReset` rather than `inflateInit`, sorry for that.

Reply:
I put the comment on the outer loop because its the only logical change, so I explained on outer loop what it should loop on